metomi/rose/date_cli.py

```diff
@@ -213,6 +217,29 @@ def _handle_old_offsets(args: list) -> list:
     return args
 
 
+def _handle_old_datetimes(args: list) -> list:
+    """Handle Legacy Rose date formats
+
+    # https://github.com/metomi/rose/issues/2589
+
+    Examples
+
+    """
+    for i, arg in enumerate(args[1:]):
+        if (
+            not (
+                args[i].startswith('--')
+                and '=' not in args[i]
+            )
+            and not arg.startswith('--')
+        ):
+            if CYLC5_FORMAT.match(arg):
+                args[i + 1] = upgrade_cylc5_datetime(arg)
+            elif UNIX_FORMAT.match(arg):
+                args[i + 1] = upgrade_unix_datetime(arg)
```

Comment:
Possibly consider this change. I found it confusing that `i` did not match the index of `args`
```suggestion
    for i, arg in enumerate(args[1:], start=1):
        if (
            not (
                args[i - 1].startswith('--')
                and '=' not in args[i - 1]
            )
            and not arg.startswith('--')
        ):
            if CYLC5_FORMAT.match(arg):
                args[i] = upgrade_cylc5_datetime(arg)
            elif UNIX_FORMAT.match(arg):
                args[i] = upgrade_unix_datetime(arg)
```